core/src/main/scala/io/finch/ToService.scala

```diff
@@ -118,11 +120,26 @@ object ToService {
             val accept =
               if (opts.negotiateContentType) req.accept.map(a => Accept.fromString(a)) else Nil
 
-            val response = out.map(oa =>
-              conformHttp(oa.toResponse(ntrA(accept), ntrE(accept)), req.version, opts)
-            )
-
-            response.run
+            val promise = Promise[Response]
+
+            (effect match {
+              case concurrent: ConcurrentEffect[F] =>
```

Comment:
I'm still not super sure we have to do that just now. I'd be happy to roll out an initial version w/o interrupts.

Reply:
I think it's low-hanging fruit and we can do it right from the beginning